packages/template-retail-react-app/app/pages/account/order-history.jsx

```diff
@@ -22,30 +22,66 @@ import {
     Img,
     Skeleton
 } from '@chakra-ui/react'
-import useCustomer from '../../commerce-api/hooks/useCustomer'
+import {useCurrentCustomer} from '../../hooks/use-current-customer'
+import {useCustomerOrders, useProduct} from 'commerce-sdk-react-preview'
 import useNavigation from '../../hooks/use-navigation'
 import {usePageUrls, useSearchParams} from '../../hooks'
-import {useAccountOrders} from './util/order-context'
 import PageActionPlaceHolder from '../../components/page-action-placeholder'
 import Link from '../../components/link'
 import {ChevronRightIcon, ReceiptIcon} from '../../components/icons'
 import Pagination from '../../components/pagination'
+import PropTypes from 'prop-types'
+
+const DEFAULT_PAGINATION_LIMIT = 10
+const ProductImage = ({productId}) => {
+    const {data: product} = useProduct({
```

Comment:
If we group the ids in the OrderHistory, and fetch the products info at once using useProducts in the OrderHistory. We can keep this ProductImage dumb by just receiving props to render the image

Reply:
Thanks, I agree we can minimize the number of network calls by using `useProducts()` hooks instead of a loop with individual product calls using `useProduct()` hook.

I confirmed that `useProducts()` returns enough product details to render the product info needed. 
I refactor the logic to use `useProducts()` for both Order history and Order details page.